packages/passport/sdk-sample-app/src/components/zkevm/EthSendTransactionExamples/SpendingCapApproval.tsx

```diff
@@ -0,0 +1,169 @@
+import React, { useCallback, useEffect, useMemo, useState } from 'react';
+import { Accordion, Form } from 'react-bootstrap';
+import { usePassportProvider } from '@/context/PassportProvider';
+import WorkflowButton from '@/components/WorkflowButton';
+import { RequestExampleProps } from '@/types';
+import { Interface } from 'ethers/lib/utils';
+import { useImmutableProvider } from '@/context/ImmutableProvider';
+import { Environment } from '@imtbl/config';
+
+const getErc20DefaultContractAddress = (environment: string) => {
+  switch (environment) {
+    case Environment.SANDBOX:
+      return '0xe9E96d1aad82562b7588F03f49aD34186f996478';
+    case Environment.PRODUCTION:
+      return '0x52a6c53869ce09a731cd772f245b97a4401d3348';
+    default:
+      return '0xba919c45c487c6a7d4e7bc5b42d4ff143a80f041';
+  }
+};
+
+function SpendingCapApproval({ disabled, handleExampleSubmitted }: RequestExampleProps) {
+  const { environment } = useImmutableProvider();
+  const iface = useMemo(() => {
+    const abi = [
+      'function approve(address spender, uint256 amount)',
+    ];
+    return new Interface(abi);
+  }, []);
+  const [fromAddress, setFromAddress] = useState<string>('');
+  const [erc20ContractAddress, setErc20ContractAddress] = useState<string>(getErc20DefaultContractAddress(environment));
+  const [spender, setSpender] = useState<string>('0xf419b5a6a3dfbf3d3a6beefff331be38ee464080');
```

Comment:
Should this default to the logged in users Passport address? I'm assuming this hardcoded value is your address?

Reply:
Lol. Yes. the intention is to ease the test process. The user only needs to click the button.
But I am willing to remove it if you think it is better to keep it blank.

Reply:
I think it makes sense to default to the logged in users Passport address, but I don't think it should default to *your* Passport address if I'm logged in 😅